soroban-env-host/src/host.rs

```diff
@@ -1895,10 +1895,50 @@ impl VmCallerEnv for Host {
 
         let min_expiration =
             self.with_ledger_info(|li| Ok(li.sequence_number.saturating_add(min.into())))?;
-        self.0.expiration_bumps.borrow_mut().0.push(LedgerBump {
-            key,
-            min_expiration,
-        });
+        self.0.expiration_bumps.borrow_mut().metered_push(
+            self,
+            LedgerBump {
+                key,
+                min_expiration,
+            },
+        )?;
+        Ok(Val::VOID)
+    }
+
+    fn bump_current_contract_instance_and_code(
+        &self,
+        _vmcaller: &mut VmCaller<Host>,
+        min: U32Val,
+    ) -> Result<Void, HostError> {
+        let min_expiration =
+            self.with_ledger_info(|li| Ok(li.sequence_number.saturating_add(min.into())))?;
+
+        let contract_id = self.get_current_contract_id_internal()?;
+        let key = self.contract_instance_ledger_key(&contract_id)?;
+        self.0.expiration_bumps.borrow_mut().metered_push(
+            self,
+            LedgerBump {
+                key: key.metered_clone(self.as_budget())?,
```

Comment:
We don't need a clone here, do we?

Reply:
We do, the same key is being used later as a reference. 
This is just an Rc bump though, so it's not really cloning anything. The `metered_clone` here is just accounting something for the Rc bump, which is really not that much. The fact that both you and @sisuresh mentioned the same thing suggests that it is not worth the trouble. I've changed this to `Rc::clone(&key)` instead (which is the Rust idiomatic way to avoid this type of confusing).